src/library/scala/collection/mutable/WrappedArray.scala

```diff
@@ -130,7 +130,7 @@ object WrappedArray {
     def update(index: Int, elem: T) { array(index) = elem }
     override def hashCode = MurmurHash3.wrappedArrayHash(array)
     override def equals(that: Any) = that match {
-      case that: ofRef[_] => Arrays.equals(array.asInstanceOf[Array[AnyRef]], that.array.asInstanceOf[Array[AnyRef]])
+      case that: ofRef[_] => that.array.canEqual(array) && array.sameElements(that.array)
```

Comment:
I think `that.array.canEqual(array)` is always return `true`.

Reply:
`Seq(1) == Seq(1.0)` also invokes `canEqual` in IterableLike which always returns true.

This code snippet is taken from GenSeqLike's equals.

Well, since it always yields true, I should remove it.